firebase-firestore/src/main/java/com/google/firebase/firestore/util/CustomClassMapper.java

```diff
@@ -932,8 +933,23 @@ private void applyFieldAnnotations(Field field) {
       }
     }
 
+    private static boolean isGetterAnnotationPresent(
+        Method method, Class<? extends Annotation> annotation) {
+      if (method.isAnnotationPresent(annotation)) {
+        return true;
+      }
+      try {
+        Method getterAnnotationsMethod =
```

Comment:
Let's add a very detailed comment on what is going on here. what this special method is and why we need it.

Reply:
Can you also add this url to your comment for reference pls?
https://github.com/JetBrains/kotlin/blob/92109a6bef9e18075996637035b63a95cf76230d/core/compiler.common.jvm/src/org/jetbrains/kotlin/load/java/JvmAbi.kt#L59

Reply:
comments updated. Thank you.